src/lavinmq/http/controller/definitions.cr

```diff
@@ -301,7 +301,7 @@ module LavinMQ
               v.exchanges.each_value.reject(&.internal?).each do |e|
                 delayed = e.arguments["x-delayed-exchange"]?
                 if delayed
-                  arguments = e.arguments.dup
```

Comment:
What type is `e.arguments` here and what it the difference between `dup` and `clone`? :)

Reply:
`e.arguments` is a [AMQ::Protocol::Table](https://github.com/cloudamqp/amq-protocol.cr/blob/main/src/amq/protocol/table.cr).

`dup` will only do a shallow copy and is defined on `Object`. `clone` is a deep copy and defined by the implementation of the class/struct.

In this case a `Table` has a `IO::Memory` and with `dup` the copy will reference the same IO which means that modifying the copy also modifies the original. `Table#clone` will create a new IO and copy the data: https://github.com/cloudamqp/amq-protocol.cr/blob/6a70d2b6857e3bdd2d69b319e57ff3cbefddaddb/src/amq/protocol/table.cr#L28-L32